android/src/com/frostwire/android/gui/fragments/preference/SearchFragment.java

```diff
@@ -141,4 +151,33 @@ private void checkAllEngines(boolean checked) {
             UIUtils.showShortMessage(getView(), R.string.search_preferences_one_engine_checked_always);
         }
     }
+
+    @Override
+    protected RecyclerView.Adapter onCreateAdapter(PreferenceScreen preferenceScreen) {
+        return new CheckedAwarePreferenceGroupAdapter(preferenceScreen, R.color.selected_search_background, R.color.basic_white);
+    }
+
+    private class CheckedAwarePreferenceGroupAdapter extends PreferenceGroupAdapter {
+
+        final int checkedDrawableId;
+        final int unCheckedDrawableId;
+
+        CheckedAwarePreferenceGroupAdapter(PreferenceGroup preferenceGroup, int checkedDrawableId, int unCheckedDrawableId) {
+            super(preferenceGroup);
+
+            this.checkedDrawableId = checkedDrawableId;
+            this.unCheckedDrawableId = unCheckedDrawableId;
+        }
+
+        @Override
+        public void onBindViewHolder(PreferenceViewHolder holder, int position) {
+            final CheckBoxPreference preference = (CheckBoxPreference) getItem(position);
+
+            preference.onBindViewHolder(holder);
+            if (!preference.getKey().equals("frostwire.prefs.search.preference_category.select_all")) {
```

Comment:
I've gone ahead and refactored this for you (sending a Pull Request)

However, I'm not sure why it's necessary to have this as a setting, as the state of this checkbox can be set programmatically on the fly if all the search engines available are checked.